src/ott/solvers/nn/losses.py

```diff
@@ -0,0 +1,85 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jax
+import jax.numpy as jnp
+from ott.geometry import pointcloud
+from ott.solvers.linear import sinkhorn
+from ott.problems.linear import linear_problem
+import ott.geometry.costs as costs
+from typing import Any, Mapping, Callable
+from types import MappingProxyType
+
+class MongeGap:
+    """
+    A class to define the Monge gap regularizer.
+    For a cost function :`math:`c` and an empirical reference :math:`\rho_x`
+    defined by samples :math:`x_i`, the (entropic) Monge gap of a vector field :math:`T` 
+    is defined as:
+        :math:`\mathcal{M}^c_{\rho_x, \epsilon} = \frac{1}{n} \sum_{i=1}^n c(x_i, T(x_i)) 
+        - W_\epsilon(\mu_x, T \# \rho_x)`.
+    
+    Args:
+    geometry_kwargs: Holds the kwargs to instanciate the geometry to compute the ``reg_ot_cost``. 
+                     Default cost function (if no cost function is passed in ``geometry_kwargs``)
+                     is the quadratic costs ``ott.geometry.costs.SqEuclidean()``.
```

Comment:
Please use links to other classes as, e.g., :class:\`~ott.geometry.costs.SqEuclidean\` or :class:\`~ott.solvers.linear.sinkhorn.Sinkhorn\`.